ast/misc.go

```diff
@@ -134,7 +134,34 @@ type ExplainStmt struct {
 
 // Restore implements Node interface.
 func (n *ExplainStmt) Restore(ctx *RestoreCtx) error {
-	return errors.New("Not implemented")
+	if showStmt, ok := n.Stmt.(*ShowStmt); ok {
+		ctx.WriteKeyWord("DESC ")
+		if err := showStmt.Table.Restore(ctx); err != nil {
+			return errors.Annotate(err, "An error occurred while restore ExplainStmt.ShowStmt.Table")
+		}
+		if showStmt.Column != nil {
+			ctx.WritePlain(" ")
+			if err := showStmt.Column.Restore(ctx); err != nil {
+				return errors.Annotate(err, "An error occurred while restore ExplainStmt.ShowStmt.Column")
+			}
+		}
+		return nil
+	}
+	ctx.WriteKeyWord("EXPLAIN ")
+	if n.Analyze {
+		ctx.WriteKeyWord("ANALYZE ")
+	} else {
+		if strings.ToLower(n.Format) == ExplainFormatDOT {
```

Comment:
Change the condition to

```suggestion
		if strings.ToLower(n.Format) != ExplainFormatROW {
```

or just let `Restore()` always write out the FORMAT. MySQL supports two entirely different formats (`JSON` and `TRADITIONAL`) and I don't think there's any reason we cannot expand the list of formats in the future.